user-console/source/org/pentaho/mantle/client/ui/xul/MantleController.java

```diff
@@ -875,6 +876,15 @@ public void setContentEditSelected( boolean selected ) {
    * Notifies currently active Javascript callback of an edit event.
    */
   public void editContentClicked() {
+    // set delay for edit/view mode switching
+    setContentEditEnabled( false );
+    Timer loadOverlayTimer = new Timer() {
+      public void run() {
+        setContentEditEnabled( true );
+      }
+    };
+    loadOverlayTimer.schedule( 1000 );
+
```

Comment:
Isn't one second a bit harsh? A user will be bound to notice the delay, especially as there will be no visual feedback while the timer runs. 

Would your fix also work if we reduce it to 250ms?